src/org/jetbrains/plugins/scala/lang/psi/impl/toplevel/typedef/StalactiteInjector.scala

```diff
@@ -0,0 +1,53 @@
+package org.jetbrains.plugins.scala.lang.psi.impl.toplevel.typedef
+
+import com.intellij.psi.PsiAnnotation
+import org.jetbrains.plugins.scala.lang.psi.api.toplevel.typedef._
+
+/**
+ * Support for https://github.com/fommil/stalactite
+ *
+ * Chooses to skip typeclass derivation (and requirements) in order to
+ * provide a much faster editing experience. Users may experience
+ * deriving failures when they run the real compiler.
+ *
+ * @author Sam Halliday
+ * @since  24/08/2017
+ */
+class StalactiteInjector extends SyntheticMembersInjector {
+  def stalactite(source: ScTypeDefinition): PsiAnnotation =
+    source.findAnnotationNoAliases("stalactite.deriving")
```

Comment:
I think you can do `source.annotations("stalactite.deriving")`, which returns a `Seq[ScAnnotation]`, which should help you with your problems in `genImplicits`